### ISSUE TITLE
`value` parameter of `createEffect` is not passed to `on` as `prevValue` when `defer` is `true`. 

### ISSUE THREAD

**Issue by vveisard** [solidjs/solid#2065]
### Describe the bug

The `value` parameter of `createEffect` is not passed to `on` as `prevValue` when `defer` is `true`. 

### Your Example Website or App

https://playground.solidjs.com/anonymous/61004170-6d09-4ce2-9582-ce966c7a131d

### Steps to Reproduce the Bug or Issue

1. Go to https://playground.solidjs.com/anonymous/61004170-6d09-4ce2-9582-ce966c7a131d
2. Observe Console tab in Result panel that `prevValue` is `undefined` for the `on` block when `defer: true`.
```
on with defer: true
    1 undefined undefined
```

### Expected behavior

As a user, I expect `value` parameter of `createEffect` to be passed to `on` as `prevValue` when `defer: true`. 

### Platform

- OS: Windows
- Browser: Microsoft Edge
- Version: 121.0.2277.98 (Official build) (64-bit)

### Additional context

Perhaps I am misunderstanding the desired behavior of `OnOptions.defer`, but this seems like a bug to me.

**Comment by ryansolid** [comment]
Yeah you are right. We return`undefined` from the first defer run but we should probably forward the `prevValue`
